react-front-end/tsrc/drm/DrmAcceptanceDialog.tsx

```diff
@@ -0,0 +1,311 @@
+ * with the License. You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Grid,
+  Theme,
+  Typography,
+} from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+import { Skeleton } from "@material-ui/lab";
+import * as E from "fp-ts/Either";
+import * as TE from "fp-ts/TaskEither";
+import { range } from "lodash";
+import { useEffect, useReducer } from "react";
+import * as React from "react";
+import * as OEQ from "@openequella/rest-api-client";
+import { pipe } from "fp-ts/function";
+import { commonString } from "../util/commonstrings";
+import { languageStrings } from "../util/langstrings";
+import { pfTernaryTypeGuard } from "../util/pointfree";
+
+const useStyles = makeStyles((theme: Theme) => ({
+  li: {
+    marginTop: theme.spacing(1),
+    marginBottom: theme.spacing(1),
+  },
+}));
+
+export interface DrmAcceptanceDialogProps {
+  /**
+   * Function fired to retrieve DRM terms from server.
+   */
+  termsProvider: () => Promise<OEQ.Drm.ItemDrmDetails>;
+  /**
+   * Function fired when the Accept button is clicked.
+   */
+  onAccept: () => void;
+  /**
+   * Function fired when the Reject button is clicked.
+   */
+  onReject: () => void;
+  /**
+   * `true` to open the dialog.
+   */
+  open: boolean;
+}
+
+// Component for non-standard DRM term which has a title and a list of terms.
+const NonStandardDrmTerms = ({
+  title,
+  terms,
+}: {
+  title: string;
+  terms: string[];
+}) => {
+  const classes = useStyles();
+  return (
+    <li className={classes.li} key={title}>
+      <Grid container direction="column">
+        <Grid item>
+          <Typography>{title}</Typography>
+        </Grid>
+        <Grid item>
+          <Grid container direction="column">
+            {terms.map((p) => (
+              <Grid item key={p}>
+                <Typography>{p}</Typography>
+              </Grid>
+            ))}
+          </Grid>
+        </Grid>
+      </Grid>
+    </li>
+  );
+};
+
+// Component for all DRM terms.
+const DrmTerms = ({
+  regularPermission,
+  additionalPermission,
+  educationSector,
+  parties,
+  customTerms,
+}: OEQ.Drm.DrmAgreements) => {
+  const classes = useStyles();
+  const standardTerms = [
+    regularPermission,
+    additionalPermission,
+    educationSector,
+  ]
+    .filter((term) => term !== undefined)
+    .map((term) => (
+      <li className={classes.li} key={term}>
+        <Grid container>
+          <Grid item>
+            <Typography>{`${term}`}</Typography>
```

Comment:
could `term` be rendered directly?

```suggestion
            <Typography>{term}</Typography>
```

Reply:
Yeah definitely!